daemon/daemon.go

```diff
@@ -1501,6 +1501,15 @@ func (daemon *Daemon) newBaseContainer(id string) *container.Container {
 	return container.NewBaseContainer(id, daemon.containerRoot(id))
 }
 
+// ResolveRepository gets the repository information from a registry service.
+func (daemon *Daemon) ResolveRepository(ref reference.Named) (*registry.RepositoryInfo, error) {
+	return daemon.registryService.ResolveRepository(ref)
+}
+
+// GetContainer returns a container identified by a reference.
+func (daemon *Daemon) GetContainer(ref string) (*container.Container, error) {
```

Comment:
[yup, agreed. it can wait.](https://github.com/docker/docker/pull/18519#discussion-diff-47036536)